Tchap/Modules/Rooms/RoomsCoordinator.swift

```diff
@@ -120,27 +121,23 @@ final class RoomsCoordinator: NSObject, RoomsCoordinatorType {
                 
                 switch response {
                 case .success(let room):
+                    // Check whether we have to consider potentially this room as a direct.
                     if thirdPartyInviteSenderID != nil {
-                        // Mark direct the new joined room if there are only 2 members
-                        self.getRoomMembersPatch(room: room, iterCount: 5) { [weak self] (response) in
-                            guard let self = self else {
-                                return
-                            }
-                            
-                            switch response {
-                            case .success(let roomMembers):
-                                if roomMembers?.members.count == 2 {
-                                    // This new joined room is a direct one
-                                    self.session.setRoom(roomID, directWithUserId: thirdPartyInviteSenderID, success: {
-                                        NSLog("[RoomsCoordinator] joinRoomByHandlingThirdPartyInvite succeeded to mark direct this new joined room")
-                                    }, failure: { _ in
-                                        NSLog("[RoomsCoordinator] joinRoomByHandlingThirdPartyInvite failed to mark direct this new joined room")
-                                    })
-                                }
-                            case .failure:
-                                NSLog("[RoomsCoordinator] joinRoomByHandlingThirdPartyInvite failed to get room members")
+                        // Check the room access rule.
+                        let rule = room.summary.tc_roomAccessRule()
+                        switch rule {
```

Comment:
To handle a single case you can write: `if case RoomAccessRule.direct = rule {`.